### PR TITLE
Optionally reprocess response

### PR DESCRIPTION
**Pull request by iloveitaly**

### CODE REVIEW

**Comment by iloveitaly** [comment]
@AlessioRocco thoughts on this?

**Comment by AlessioRocco** [comment]
@iloveitaly I like that idea but instead to add the configuration for the whole application maybe is better to  add it at `Job` level so the developer can choose which must reprocess the response and which don't, in case you like this way we have to decide the default behavior, if reprocess by default or not.

**Comment by iloveitaly** [comment]
@AlessioRocco Good idea! I'll change this up.

**Comment by iloveitaly** [comment]
@AlessioRocco Fixed this one up! Lmk what you think :)

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -39,6 +39,7 @@
   # reset config before each spec
   config.before(:each) do
     Rails.configuration.cangaroo.basic_auth = false
+    Rails.configuration.cangaroo.process_response = true
```
Comment (AlessioRocco) [inline spec/rails_helper.rb]: I think this isn't needed anymore

File: app/jobs/cangaroo/job.rb [hunk 2]
```diff
@@ -31,6 +32,10 @@ def restart_flow(response)
       # if no json was returned, the response should be discarded
       return if response.blank?
 
+      if !self.process_response
```
Comment (AlessioRocco) [inline app/jobs/cangaroo/job.rb]: Why do you not use `return unless self.process_response`? 
Personally I prefer to follow the [ruby-style-guide](https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)

**Comment by iloveitaly** [comment]
@AlessioRocco Fixed up this PR based on your comments!

**Comment by iloveitaly** [comment]
Friendly ping @AlessioRocco

**Comment by AlessioRocco** [comment]
@iloveitaly sorry for the delay, merged.

**Pull request merged**